File: Java/FlattenDirectory.java
// FlattenDirectory.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardCopyOption;

public class FlattenDirectory {
    public static void flattenDirectory(String sourceDir, String targetDir) {
        File source = new File(sourceDir);
        File target = new File(targetDir);

        if (!source.exists() || !source.isDirectory()) {
            System.err.println("Error: Source directory does not exist or is not a directory.");
            System.exit(1);
        }

        if (!target.exists()) {
            target.mkdirs();
        }

        flatten(source, target);
        System.out.println("All files have been extracted to " + targetDir + " without preserving folder structure.");
    }

    private static void flatten(File sourceDir, File targetDir) {
        for (File entry : sourceDir.listFiles()) {
            if (entry.isFile()) {
                try {
                    Files.copy(entry.toPath(), Path.of(targetDir.getPath(), entry.getName()), StandardCopyOption.REPLACE_EXISTING);
                } catch (IOException e) {
                    System.err.println("Error copying file: " + entry.getPath());
                }
            } else if (entry.isDirectory()) {
                flatten(entry, targetDir);
            }
        }
    }

    public static void main(String[] args) {
        if (args.length != 2) {
            System.out.println("Usage: java FlattenDirectory <source_directory> <target_directory>");
            System.exit(1);
        }
        flattenDirectory(args[0], args[1]);
    }
}

File: cs/FlattenDirectory.cs
using System;
using System.IO;

class FlattenDirectory
{
    static void FlattenDirectoryStructure(string sourceDir, string targetDir)
    {
        if (!Directory.Exists(sourceDir))
        {
            Console.Error.WriteLine("Error: Source directory does not exist.");
            Environment.Exit(1);
        }

        if (!Directory.Exists(targetDir))
        {
            Directory.CreateDirectory(targetDir);
        }

        foreach (var filePath in Directory.EnumerateFiles(sourceDir, "*", SearchOption.AllDirectories))
        {
            var fileName = Path.GetFileName(filePath);
            var targetPath = Path.Combine(targetDir, fileName);
            File.Copy(filePath, targetPath, true);
        }
        Console.WriteLine($"All files have been extracted to {targetDir} without preserving folder structure.");
    }

    static void Main(string[] args)
    {
        if (args.Length != 2)
        {
            Console.WriteLine("Usage: dotnet FlattenDirectory.dll <source_directory> <target_directory>");
            Environment.Exit(1);
        }
        FlattenDirectoryStructure(args[0], args[1]);
    }
}
